--- /tmp/cairo/cairo.scm	2014-11-29 14:54:43.000000000 +0100
+++ /home/kooda/sources/chicken/cairo/cairo.scm	2014-11-28 21:22:12.000000000 +0100
@@ -361,8 +361,10 @@
 ; ------------------------------------------------
 
 (define cairo-stroke (foreign-lambda void "cairo_stroke" cairo_t))
+(define cairo-stroke-preserve (foreign-lambda void "cairo_stroke_preserve" cairo_t))
 
 (define cairo-fill (foreign-lambda void "cairo_fill" cairo_t))
+(define cairo-fill-preserve (foreign-lambda void "cairo_fill_preserve" cairo_t))
 
 ; ------------------------------------------------
 
